lib/Dialect/TTMetal/Transforms/Passes.cpp

```diff
@@ -272,34 +272,42 @@ class TTIRToTTMetalLayoutRewriter : public OpRewritePattern<ttir::ToLayoutOp> {
     tensixBlock->addArgument(inputCBTy, op.getLoc());
     tensixBlock->addArgument(outputCBTy, op.getLoc());
 
-    int shardTileVolume = 1;
-    for (auto dim : (shouldTilize ? outputLayout.getMemref().getShape()
-                                  : inputLayout.getMemref().getShape())) {
-      shardTileVolume *= dim;
-    }
+    llvm::ArrayRef<int64_t> shardTileShape =
+        (shouldTilize ? outputLayout.getMemref().getShape()
+                      : inputLayout.getMemref().getShape());
+
+    assert(shardTileShape.size() == 2 && "Tile shape rank must be 2");
```

Comment:
I don't think this is true, I think it's safe to assert that it's at least 2 since 1D data is kind of awkward to support, impossible if tiled.  But the memref shape == tensor grid shape which can be rank N.

Reply:
We need a test for it for sure, currently I think we should try and get the skeleton of a bunch of the core pieces in place.  Then we can go wider with testing